src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -389,11 +389,22 @@ private void EnqueueRequest(Message message, ActivationData targetActivation)
                 RejectMessage(message, Message.RejectionTypes.Overloaded, overloadException, "Target activation is overloaded " + targetActivation);
                 return;
             }
-            
-            bool enqueuedOk = targetActivation.EnqueueMessage(message);
-            if (!enqueuedOk)
-            {
-                ProcessRequestToInvalidActivation(message, targetActivation.Address, targetActivation.ForwardingAddress, "EnqueueRequest");
+
+            switch (targetActivation.EnqueueMessage(message))
+            {
+                case ActivationData.EnqueueMessageResult.Success:
+                    // Great, nothing to do
+                    break;
+                case ActivationData.EnqueueMessageResult.ErrorInvalidActivation:
+                    ProcessRequestToInvalidActivation(message, targetActivation.Address, targetActivation.ForwardingAddress, "EnqueueRequest");
+                    break;
+                case ActivationData.EnqueueMessageResult.ErrorStuckActivation:
+                    // Avoid any new call to this activation
+                    catalog.RemoveFromDirectory(targetActivation).Wait();
```

Comment:
I know this is annoying, but the `.Wait()` is a little worrisome here, since it can involve making a remote call.
It just bears a little discussion whether or not we want to do it this way or make the entire call chain async.

Maybe the answer is to make it block for now and change it to be async when `ValueTask<T>` is available so that we don't incur the Task overhead